testonly/hammer/hammer.go

```diff
@@ -221,6 +224,18 @@ type hammerState struct {
 	valueIdx int
 }
 
+type safePRNG struct {
```

Comment:
If you generate one PRNG per thread, do you need locking?

Reply:
Technically not, but I think it's safer in case there's ever a change in future that has more than one goroutine doing things on a single map.

Reply:
Hmm, I wish there was a way to do "this thing can only be accessed in this go routine". Do you think the structs could be arranged in such a way as to make that possible?

Even with the mutex, a situation where multiple threads access the same object would still cause difficulties because of the non deterministic order of calls. It seems like what we really want is a go routine wide lock on the object, not a per-call lock.

Reply:
Ah, good point -- dropped the mutex protection.